src/styles/button.scss

```
@@ -60,215 +46,32 @@ $block: #{$fd-namespace}-button;
-    @include positive();
-
-    &.#{$block}--toggled {
-      @include positiveToggled();
-    }
-  }
-
-  &.#{$block}--negative {
-    @include negative();
-
-    &.#{$block}--toggled {
-      @include negativeToggled();
-    }
-  }
-
-  &.#{$block}--attention {
-    @include attention();
-
-    &.#{$block}--toggled {
-      @include attentionToggled();
-    }
-  }
-
-  &.#{$block}--emphasized {
-    font-weight: var(--sapButton_Emphasized_FontWeight);
-    border-width: var(--fdButton_Emphasized_Border_Width);
-
-    @include emphasized();
-
-    &.#{$block}--toggled {
-      @include standardToggled();
-    }
-  }
-
-  &.#{$block}--transparent {
-    @include transparent();
-
-    &.#{$block}--toggled {
-      @include standardToggled();
-    }
-  }
-}
-
-@mixin fd-button-focus-state() {
-  @include fd-button-focus(var(--fdButton_Outline_Offset));
-
-  box-shadow: none;
-
-  &.#{$block}--emphasized {
-    outline-color: var(--sapContent_ContrastFocusColor);
-
-    &.#{$block}--toggled {
-      outline-color: var(--fdButton_Outline_Contrast);
-    }
-  }
-}
-
-@mixin fd-button-pressed-state() {
-  @include standardPressed();
-
-  box-shadow: none;
-  outline: none;
-
-  &.#{$block}--toggled {
-    @include standardToggledHoverPressed();
-  }
-
-  &.#{$block}--positive {
-    @include positivePressed();
-
-    &.#{$block}--toggled {
-      @include positiveToggledHoverPressed();
-    }
-  }
-
-  &.#{$block}--negative {
-    @include negativePressed();
-
-    &.#{$block}--toggled {
-      @include negativeToggledHoverPressed();
-    }
-  }
-
-  &.#{$block}--attention {
-    @include attentionPressed();
-
-    &.#{$block}--toggled {
-      @include attentionToggledHoverPressed();
-    }
-  }
 
-  &.#{$block}--emphasized {
-    @include emphasizedPressed();
-
-    &.#{$block}--toggled {
-      @include standardToggledHoverPressed();
-    }
-  }
-
-  &.#{$block}--transparent {
-    @include transparentPressed();
-
-    &.#{$block}--toggled {
-      @include standardToggledHoverPressed();
-    }
-  }
-}
-
-@mixin fd-button-hover-state() {
-  @include standardHover();
-
-  &.#{$block}--toggled {
-    @include standardToggledHoverPressed();
-  }
-
-  &.#{$block}--positive {
-    @include positiveHover();
-
-    &.#{$block}--toggled {
-      @include positiveToggledHoverPressed();
-    }
-  }
-
-  &.#{$block}--negative {
-    @include negativeHover();
-
-    &.#{$block}--toggled {
-      @include negativeToggledHoverPressed();
-    }
-  }
-
-  &.#{$block}--attention {
-    @include attentionHover();
-
-    &.#{$block}--toggled {
-      @include attentionToggledHoverPressed();
-    }
-  }
-
-  &.#{$block}--emphasized {
-    @include emphasizedHover();
+  &__text {
+    @include fd-reset();
+    @include fd-ellipsis();
 
-    &.#{$block}--toggled {
-      @include standardToggledHoverPressed();
-    }
+    font-weight: inherit;
+    color: inherit;
   }
 
-  &.#{$block}--transparent {
-    @include transparentHover();
+  * + * {
```

```suggestion
  *:not(:first-child) {
```
It's a longer, but easier to grasp